internal/cmd/lsp-record/lsp-record.go

```diff
@@ -401,21 +405,48 @@ func test() error {
 	return nil
 }
 
+func exists(path string) bool {
+	if _, err := os.Stat(path); os.IsNotExist(err) {
+		return false
+	} else if err != nil {
+		log.Fatalf("unexpected error testing existance of %s: %s", path, err)
+	}
+	return true
+}
+
+func fileCreateNotExist(path string) (*os.File, error) {
+	if exists(path) {
+		return nil, errors.Errorf("%s already exists", path)
+	}
+	return os.Create(path)
+}
+
 func mainErr() error {
 	flag.Usage = usage
 	flag.Parse()
 	args := flag.Args()
-	if len(args) != 2 || (args[0] != "record" && args[0] != "test") {
+	if (len(args) != 2 && len(args) != 3) || (args[0] != "record" && args[0] != "test") {
 		flag.Usage()
 		os.Exit(2)
 	}
 	action := args[0]
 	lang := args[1]
+	testcase := "-"
+	if len(args) == 3 {
+		testcase = args[2]
+	}
+
 	image := fmt.Sprintf("sgtest/codeintel-%s", lang)
+	testdataDir := filepath.Join("dockerfiles", lang, "testdata")
 	dockerfile := filepath.Join("dockerfiles", lang, "Dockerfile")
-	if _, err := os.Stat(dockerfile); os.IsNotExist(err) {
+	if !exists(dockerfile) {
 		return errors.Errorf("%s could not be found. Ensure you are running from github.com/sourcegraph/lsp-adapter directory and that %s integration exists", dockerfile, lang)
 	}
+	if testcase != "-" && !exists(testdataDir) {
+		if err := os.MkdirAll(testdataDir, os.ModePerm); err != nil {
+			return err
```

Comment:
Can we use `errors.Wrap` here? I'm not sure if `MkdirAll` will return a descriptive enough error on its own to help us realize what happened